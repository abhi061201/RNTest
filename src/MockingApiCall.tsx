import {StyleSheet, Text, View} from 'react-native';
import React, {useEffect, useState} from 'react';

interface User {
  id: string;
  name: string;
}
const MockingApiCall = () => {
  const [user, setUser] = useState<User[]>([]);
  const fetchUsers = async () => {
    const response = await fetch('https://jsonplaceholder.typicode.com/users');
    const data = await response.json();
    setUser(data);
  };
  useEffect(() => {
    fetchUsers();
  }, []);
  return (
    <View>
      <Text>Users List:</Text>
      {user.map(user => (
        <Text key={user.id}>{user.name}</Text>
      ))}
    </View>
  );
};

export default MockingApiCall;

const styles = StyleSheet.create({});
