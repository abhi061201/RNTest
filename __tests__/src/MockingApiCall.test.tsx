import React from 'react';
import {render, waitFor} from '@testing-library/react-native';
import MockingApiCall from '../../src/MockingApiCall';

// Mock global fetch function
global.fetch = jest.fn(() =>
  Promise.resolve({
    json: () =>
      Promise.resolve([
        {id: 1, name: 'John Doe'},
        {id: 2, name: 'Jane Smith'},
      ]),
  }),
) as jest.Mock;

test('renders UserList correctly', async () => {
  const {toJSON, getByText} = render(<MockingApiCall />);

  // Wait for the API response to be processed
  // this is just for checking k api call done or not
  await waitFor(() => getByText('John Doe'));

  // Take a snapshot
  expect(toJSON()).toMatchSnapshot();
});
