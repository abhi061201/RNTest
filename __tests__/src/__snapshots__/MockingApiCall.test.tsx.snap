// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders UserList correctly 1`] = `
<View>
  <Text>
    Users List:
  </Text>
  <Text>
    John Doe
  </Text>
  <Text>
    Jane Smith
  </Text>
</View>
`;
